DOCS-7112 Static Analysis Doc Set Update to Code Analysis (#21501)
DataDog/documentation: content/en/code_analysis/_index.md

  text: "Monitor all your CI pipelines with Datadog"
- link: "/integrations/guide/source-code-integration/"
  tag: "Documentation"
  text: "Learn about the Source Code Integration"
- link: "/code_analysis/static_analysis"
  tag: "Documentation"
  text: "Learn about Static Analysis"
- link: "/code_analysis/software_composition_analysis"
  tag: "Documentation"
  text: "Learn about Software Composition Analysis"
---

{{% site-region region="gov" %}}
<div class="alert alert-danger">
    Code Analysis is not available for the {{< region-param key="dd_site_name" >}} site.
</div>
{{% /site-region %}}

{{< callout url="#" btn_hidden="true" header="Try the Beta!" >}}
Code Analysis is in public beta.
{{< /callout >}}

## Overview

Code Analysis is composed of the [Static Analysis][1] and [Software Composition Analysis (SCA)][2] products. 

Static Analysis
: Scans your bespoke code for maintainability issues, bugs, performance issues, and security vulnerabilities early in the development lifecycle to catch issues from reaching production and, when possible, provide suggested fixes to help engineering teams address these issues before they impact users.

Software Composition Analysis 

> : Scans the open-source libraries that are imported into your repositories for known vulnerabilities. Coming soon will be the ability to block libraries with certain licenses and suggestions on which version of a library you should update to so that you can resolve the vulnerability.

Reviewer: ```suggestion
: Scans the open source libraries that are imported into your repositories for known vulnerabilities. Coming soon will be the ability to block libraries with certain licenses and suggestions on which version of a library you should update to so that you can resolve the vulnerability.
```

Sorry I missed this the first time!

> : Scans the open-source libraries that are imported into your repositories for known vulnerabilities. Coming soon will be the ability to block libraries with certain licenses and suggestions on which version of a library you should update to so that you can resolve the vulnerability.

Reviewer: ```suggestion
: Scans the open-source libraries that are imported into your repositories for known vulnerabilities. In a future release, you will have the ability to block libraries with certain licenses, and apply suggestions to upgrade to remediated library versions, so that you can resolve the vulnerability.
```
I know we typically avoid words as "coming soon" , I think we should change this to "in a future release" ?

Reviewer: Can you fix open-source here too, to "open source" ?

Once you have configured Code Analysis, you can head to the [Code Analysis page][9] to see a summary of the results from the Static Analysis and SCA scans for each of your configured repositories. The summarized results will always be for the latest commit on the default branch of each repository to ensure you are seeing all the existing problems on each repository that you may want to triage and fix.

> Select a repository from the list to search through and manage violations in for that specific repository. By default, the results will be filtered to the latest commit on the default branch of the repository, but you may change the branch or commit at the top of the page if desired. Regardless of the selected branch or commit, all results will be organized into the following views:

Reviewer: ```suggestion
Select a repository from the list to search through and manage violations for that specific repository. By default, the results are filtered to the latest commit on the default branch of the repository, but you may change the branch or commit at the top of the page. Regardless of the selected branch or commit, all results will be organized into the following views:
```

{{< tabs >}}
{{% tab "Code Vulnerabilities" %}}

{{< img src="code_analysis/shopist_code_vulnerabilities.png" alt="Code vulnerabilities on the Code Analysis page for the Datadog Shopist service and repository" style="width:100%;">}}

Identify and address code security risks detected by Static Analysis in the **Code Vulnerabilities** view.

{{% /tab %}}
{{% tab "Code Quality" %}}

{{< img src="code_analysis/shopist_code_quality.png" alt="Code quality vulnerabilities on the Code Analysis page for the Datadog Shopist service and repository" style="width:100%;">}}

Identify and address poor coding practices detected by Static Analysis in the **Code Quality** view.

{{% /tab %}}
{{% tab "Library Vulnerabilities" %}}

{{< img src="code_analysis/shopist_lib_vulnerabilities.png" alt="Library vulnerabilities on the Code Analysis page for the Datadog Shopist service and repository" style="width:100%;">}}

> Identify and address vulnerable open-source libraries detected by SCA in the **Library Vulnerabilities** view.

Reviewer: ```suggestion
Identify and address vulnerable open source libraries detected by SCA in the **Library Vulnerabilities** view.
```

further_reading:
- link: "https://www.datadoghq.com/blog/monitor-ci-pipelines/"
  tag: "Blog"
  text: "Monitor all your CI pipelines with Datadog"
- link: "/integrations/guide/source-code-integration/"
  tag: "Documentation"
  text: "Learn about the Source Code Integration"
- link: "/code_analysis/static_analysis"
  tag: "Documentation"
  text: "Learn about Static Analysis"
---

## Overview

{{% site-region region="gov" %}}
<div class="alert alert-danger">
    Code Analysis is not available for the {{< region-param key="dd_site_name" >}} site.
</div>
{{% /site-region %}}

Code Analysis is Static Analysis and Software Composition Analysis (SCA) combined.

You can search and manage code violations and library dependencies for any service in Datadog.

Code Vulnerabilities
: Identify and address risks in code security.

Code Quality
: Identify and address poor practices in code quality.

Library Vulnerabilities
: Identify and address security in libraries.

Library List
: Identify and remove prohibited libraries.

## Setup

{{< tabs >}}
{{% tab "Static Analysis" %}}

To run Static Analysis, [see the documentation][101].

[101]: /code_analysis/static_analysis

{{% /tab %}}

> {{< /tabs >}}

Author: We can use the list formatting with a partial like: 

```
{{< whatsnext desc="You can explore the following out-of-the-box RUM dashboards:" >}}
  {{< nextlink href="/real_user_monitoring/dashboards/performance" >}}<u>Performance Overviews</u>: See a global view of your website/app performance and demographics. {{< /nextlink >}}
  {{< nextlink href="/real_user_monitoring/dashboards/testing_and_deployment" >}}<u>Testing and Deployment</u>: Evaluate your browser tests' application coverage and identify popular elements in your application to track using RUM and Synthetics data. {{< /nextlink >}}
  {{< nextlink href="/real_user_monitoring/dashboards/usage" >}}<u>Usage</u>: Analyze user session and usage data for your RUM applications, including frustration signals. {{< /nextlink >}}
  {{< nextlink href="/real_user_monitoring/dashboards/errors" >}}<u>Errors</u>: Observe errors that appear in user consoles by browser and device type. {{< /nextlink >}}
{{< /whatsnext >}}
```

---
title: Code Analysis
kind: documentation
description: Learn about Datadog Code Analysis.
is_beta: true
further_reading:
- link: "https://www.datadoghq.com/blog/monitor-ci-pipelines/"
  tag: "Blog"
  text: "Monitor all your CI pipelines with Datadog"
- link: "/integrations/guide/source-code-integration/"
  tag: "Documentation"
  text: "Learn about the Source Code Integration"
- link: "/code_analysis/static_analysis"
  tag: "Documentation"
  text: "Learn about Static Analysis"
---

## Overview

{{% site-region region="us,us3,us5,eu,ap1" %}}
<div class="alert alert-warning">

>   Static Analysis is in private beta. Python, Java, C#, JavaScript, TypeScript, and Docker are the only supported languages. To request access, <a href="/help">contact Support</a>.

Reviewer 2: Should we be updating the Static Analysis text in here?
